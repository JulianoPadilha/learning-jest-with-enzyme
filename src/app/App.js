import React, { Component } from 'react';

// const Test = () => <div>Testing</div>

class App extends Component {
  render() {
    return (
      <div>
        <h1 className="app">React App for testing with Jest and Ezyme</h1>
        <ul>
          <li>Test 1</li>
          <li>Test 2</li>
          <li>Test 3</li>
        </ul>
        {/* <Test /> */}
      </div>
    )
  }
}

export default App;