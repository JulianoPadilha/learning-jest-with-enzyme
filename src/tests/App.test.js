import React from 'react';
import ReactDOM from 'react-dom';
import App, { Link } from '../app/App';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<App /> shallow rendering', () => {
  it('should render App', () => {
    const wrapper = shallow(<App />, { context: {}, disableLifecycleMethods: true });
    console.log(wrapper.debug());
  });

  it('should contain elements', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('h1').length).toBe(1);
    expect(wrapper.find('.app').exists()).toBe(true);
    expect(wrapper.find('ul').children().length).toBe(3);
    expect(wrapper.find('h1').text()).toBe('React App for testing with Jest and Ezyme');
  });

  it('matches the snapshot', () => {
    const wrapper = render(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('<Link />', () => {
  it('link component accepts address prop', () => {
    const wrapper = shallow(<Link address='www.google.com' />);
    // console.log(wrapper.instance());
    expect(wrapper.instance().props.address).toBe('www.google.com');
  });

  it('a tag node renders href correctly', () => {
    const wrapper = shallow(<Link address='www.google.com' />);
    expect(wrapper.props().href).toBe('www.google.com');
  });

  it('returns null with true hide prop', () => {
    const wrapper = shallow(<Link hide={false} />);
    expect(wrapper.find('a').length).toBe(1);
  });

  it('returns null with true hide prop', () => {
    const wrapper = shallow(<Link hide={false} />)
    expect(wrapper.find('a').length).toBe(1);
    wrapper.setProps({ hide: true });
    expect(wrapper.get(0)).toBeNull();
  });
})