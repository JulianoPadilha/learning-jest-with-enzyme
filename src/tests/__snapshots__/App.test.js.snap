// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div>
  <h1
    class="app"
  >
    React App for testing with Jest and Ezyme
  </h1>
  <ul>
    <li>
      Test 1
    </li>
    <li>
      Test 2
    </li>
    <li>
      Test 3
    </li>
  </ul>
</div>
`;

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <div>
    <h1
      className="app"
    >
      React App for testing with Jest and Ezyme
    </h1>
    <ul>
      <li>
        Test 1
      </li>
      <li>
        Test 2
      </li>
      <li>
        Test 3
      </li>
    </ul>
  </div>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div>
  <h1
    class="app"
  >
    React App for testing with Jest and Ezyme
  </h1>
  <ul>
    <li>
      Test 1
    </li>
    <li>
      Test 2
    </li>
    <li>
      Test 3
    </li>
  </ul>
</div>
`;
